src/include/storage/store/csr_node_group.h

```diff
@@ -122,16 +122,23 @@ struct CSRNodeGroupScanState final : NodeGroupScanState {
     // States at the csr list level. Cached during scan over a single csr list.
     CSRNodeGroupScanSource source = CSRNodeGroupScanSource::COMMITTED_PERSISTENT;
 
-    CSRNodeGroupScanState(MemoryManager& memoryManager, common::idx_t numChunks)
-        : NodeGroupScanState{numChunks},
-          csrHeader{std::make_unique<ChunkedCSRHeader>(memoryManager, false,
-              common::StorageConstants::NODE_GROUP_SIZE, ResidencyState::IN_MEMORY)} {}
+    explicit CSRNodeGroupScanState(common::idx_t numChunks)
+        : NodeGroupScanState{numChunks}, csrHeader{nullptr} {}
+
+    void initCSRHeader(MemoryManager& mm) {
+        if (!csrHeader) {
```

Comment:
Normally this is not good practice. If u can have the assumption that csrHeader is only initialized once, then this if check shouldn't be needed.